src/test/java/BaseTest.java

```diff
@@ -1,18 +1,134 @@
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(60));
+        driver.manage().window().maximize();
+
+        url = BaseURL;
+        driver.get(url);
+
+
+    }
+
+    @AfterMethod
+    public static void closeBrowser() {
+        driver.quit();
+    }
+
+    //Methods for logging in
+    public static void navigateToPage(String url) {
+        driver.get(url);
+    }
+
+    public static void provideEmail(String email) {
+        WebElement emailField = driver.findElement(By.cssSelector("[type = 'email']"));
+        emailField.sendKeys(email);
+    }
+
+    public static void providePassword(String password) {
+        WebElement passwordField = driver.findElement(By.cssSelector("[type = 'password']"));
+        passwordField.sendKeys(password);
+    }
+
+    public static void clickSubmit() {
+        WebElement submitButton = driver.findElement(By.cssSelector("button[type = 'submit']"));
+        submitButton.click();
+    }
+
+
+    //Methods for playing a song
+    public void enterAllSongs() {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(4));
+        WebElement allSongs = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("a[href='#!/songs']")));
+        allSongs.click();
+
+        // WebElement allSongs = driver.findElement(By.cssSelector("a[href='#!/songs']"));
+        //allSongs.click();
+
+        //Thread.sleep(1000);
+        // allSongs.click();
+        //
+    }
+
+    public void selectSong() {
+        WebElement song = driver.findElement(By.xpath("//tr[@class='song-item']"));
+        song.click();
+
+        //throws InterruptedException
+        //WebElement song = driver.findElement(By.xpath("//tr[@class='song-item']"));
+        //Thread.sleep(1000);
+
+    }
+
+    public void enterButtonPlaySong() {
+        WebElement buttonPlaySong = driver.findElement(By.xpath("//span[@title='Play or resume']"));
+        Actions actions = new Actions(driver);
+        actions.click(buttonPlaySong).perform();
+    }
+
+
+    public boolean isDisplayedPlayingSong() {
+
+        WebElement songIsPlaying = driver.findElement(By.cssSelector("[data-testid = 'sound-bar-play']"));
+        return songIsPlaying.isDisplayed();
+
+
+    }
+
+    //additional things for Login test
+    public boolean isHomePageElementDisplayed() {
+        WebElement homePage = driver.findElement(By.xpath("/html/body/div/div/div/nav/section[1]/ul/li[1]/a"));
+        return homePage.isDisplayed();
+
+    }
+
+    @DataProvider(name = "incorrectLoginData")
+    public Object[][] getDataProviders() {
+        return new Object[][]{{"invalid@gmail.com", "invalidPass"}, {"onlyEmail@gmail.com", ""}, {"", ""}
+
+        };
+    }
+
+    Actions renamePlaylist() = new Actions(driver);
+
+    WebElement element = driver.findElement(By.xpath("/html/body/div/div/div/nav/section[2]/ul/li[3]/a"));
+renamePlaylist.doubleClick(element).
+
+    perform();
+
+
 }
+
+
```

Comment:
The code is not compiled! Please finish the homework!You got this problem, please correct it. 

![image](https://user-images.githubusercontent.com/104761482/223487653-724666c6-a54e-45df-9ff6-d8669ff7ceee.png)

________________________________________________________________________________________________________________________

**Please delete the empty line!** An Empty Line is a "Code Smell"
The short answer is, _They’re more an annoyance than a problem.They won’t actually break anything—which is why your code continues to run—but they make it awkward to read and change your code, especially if you’re dealing with multiple line breaks per line of real code. They also add just a tiny bit of size to your files, which is nice to reduce._

__________________________________________________________________________________________________________________________

How can I beautify code in IntelliJ:

The formatting shortcuts in Intellij IDEA are :
For Windows : Ctrl + Alt + L.
For Ubuntu : Ctrl + Alt + Windows + L.
For Mac : ⌥ (Option) + ⌘ (Command) + L.

![image](https://user-images.githubusercontent.com/104761482/209774134-9e63bfed-d768-4455-a860-028305c85e17.png)